src/libostree/ostree-repo-commit.c

```diff
@@ -509,6 +509,33 @@ _ostree_repo_bare_content_commit (OstreeRepo                 *self,
 {
   OstreeRealRepoBareContent *real = (OstreeRealRepoBareContent*) barewrite;
   g_assert (real->initialized);
+
+  if ((self->min_free_space_percent > 0 || self->min_free_space_mb > 0) && self->in_transaction)
+    {
+      g_mutex_lock (&self->txn_lock);
```

Comment:
Ca we can move the locking to after the `fstat`?

Reply:
Ah yes, makes sense.